web/components/roomConfig/GitlabRepoConfig.tsx

```diff
@@ -0,0 +1,167 @@
+                if (res[0]) {
+                    setFilter({instance: res[0].name});
+                }
+            } else {
+                setResults(res as GitLabRepoConnectionProjectTarget[]);
+            }
+        });
+    }, [api, filter]);
+
+    const updateSearchFn = useCallback((evt: InputEvent) => {
+        setFilter({...filter, search: (evt.target as HTMLInputElement).value });
+    }, [filter, setFilter]);
+
+    useEffect(() => {
+        if (debounceTimer) {
+            clearTimeout(debounceTimer);
+        }
+        // Browser types
+        setDebounceTimer(setTimeout(searchFn, 500) as unknown as number);
+        // Things break if we depend on the thing we are clearing.
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [setDebounceTimer, searchFn, filter, instances]);
+
+    useEffect(() => {
+        const hasResult = results?.find(n => n.name === filter.search);
+        if (hasResult) {
+            onPicked(hasResult.state);
+            setCurrentProjectPath(hasResult.state.path);
+        }
+    }, [onPicked, results, filter, setCurrentProjectPath]);
+
+
+
+    return <div>
+        {instances === null && <p> Loading GitLab instances. </p>}
+        {instances?.length === 0 && <p> You are not logged into any GitLab instances. </p>}
+        <InputField visible={instances?.length > 0} label="GitLab Instance" noPadding={true}>
+            {/*TODO: on change required here*/}
+            <select>
+                {instances?.map(i => <option key={i.name}>{i.name}</option>)}
+            </select>
+        </InputField>
+        <InputField visible={instances?.length > 0} label="Project" noPadding={true}>
+            <small>{currentProjectPath ?? ""}</small>
+            <input onChange={updateSearchFn} list="gitlab-projects" type="text" />
+            <datalist id="gitlab-projects">
+                {results?.map(i => <option path={i.state.path} value={i.name} key={i.name} />)}
+            </datalist>
+        </InputField>
+    </div>;
+}
+
+const ConnectionConfiguration: FunctionComponent<ConnectionConfigurationProps<unknown, GitLabRepoResponseItem, GitLabRepoConnectionState>> = ({api, existingConnection, onSave, onRemove }) => {
+    const [ignoredHooks, setIgnoredHooks] = useState<string[]>(existingConnection?.config.ignoreHooks || []);
+    const setIgnoredHook = useCallback(evt => {
+        const key = (evt.target as HTMLElement).getAttribute('x-event-name');
+        if (ignoredHooks.includes(key)) {
+            setIgnoredHooks(ignoredHooks.filter(k => k !== key));
+        } else {
+            setIgnoredHooks([...ignoredHooks, key]);
+        }
+    }, [ignoredHooks, setIgnoredHooks]);
+    const [newInstanceState, setNewInstanceState] = useState<GitLabRepoConnectionState|null>(null);
+
+
+
+    const canEdit = !existingConnection || (existingConnection?.canEdit ?? false);
+    const commandPrefixRef = createRef<HTMLInputElement>();
+
+    const onSaveCb = useCallback(() => {
```

Comment:
Just a suggestion for naming these:
For several years I've adopted the following naming scheme.

`onSave` – the prop from a parent component
`handleSave` – the callback in the component (and likely calls `onSave`)

Reply:
I like that, will use it.